src/pages/api/frontend/localhost-graphql-fetch.ts

```diff
@@ -0,0 +1,31 @@
+import { GraphQLClient } from 'graphql-request'
+import type { NextApiRequest, NextApiResponse } from 'next'
+
+import { endpoint } from '@/api/endpoint'
+import { ParsedArgs } from '@/api/graphql-fetch'
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  const { query, variables } = JSON.parse(req.body as string) as ParsedArgs
+
+  if (!req.headers.cookie)
+    res.status(403).json({ message: 'No auth cookie provided!' })
+
+  function executeQuery() {
+    const client = new GraphQLClient(endpoint, {
+      credentials: 'include',
+      headers: { Cookie: req.headers.cookie! },
+    })
```

Comment:
Very good. I think, we will need to take a look at it again when it is in kratos stuff in api is deployed in staging. I'm uneasy with the fact, that we send only two headers, but indeed, testing locally, that worked while `headers: req.headers` haven't.

Reply:
what headers might be missing compared to the client request?